src/components/map/EsriMappingUtil.tsx

```diff
@@ -0,0 +1,74 @@
+import mapboxgl from "mapbox-gl";
+import httpClient from "../../httpClient";
+
+// Params: url - to an esri vector tile service
+// Returns: Modified style object with attributes for Mapbox GL JS compatability 
+async function prepare(url: string) {
+
+  const styleUrl = url + "/resources/styles/root.json";
+
+  const api = new httpClient()
+
+  let fetchedStyle = await api.getStyles(styleUrl);
+  fetchedStyle.sprite = url + "/resources/sprites/sprite";
+  fetchedStyle.glyphs = url + "/resources/fonts/{fontstack}/{range}.pbf";
+  fetchedStyle.sources.esri = {
+    type: 'vector',
+    tiles: [url + "/tile/{z}/{y}/{x}.pbf"],
+    maxzoom: 23,
+  };
+
+  return fetchedStyle;
+};
+
+// Params: url - to an esri vector tile service
+// Returns: Mapbox style object with applied serotracker styling expressions
+export async function getEsriVectorSourceStyle(url: string) {
+    let style = await prepare(url)
+
+    var source = style.sources.esri as mapboxgl.VectorSource;
+
+    const l = style.layers[0] as mapboxgl.Layer;
+    if (l.id === "Countries") {
+      source.promoteId = { "Countries": "CODE" }
+      l.paint = {
+        'fill-outline-color' : "#455a64",
```

Comment:
suggestion: could store these colour hex codes in some sort of constants file. would make things more readable if we ever want to either change these colours in the future or reuse them somewhere else in the app